x-pack/plugins/reporting/server/config/index.ts

```diff
@@ -21,7 +21,7 @@ export const config: PluginConfigDescriptor<ReportingConfigType> = {
     unused('capture.viewport', { level: 'warning' }), // deprecated as unused since 7.16
     (settings, fromPath, addDeprecation) => {
       const reporting = get(settings, fromPath);
-      if (reporting?.roles?.enabled !== false) {
+      if (reporting?.roles?.enabled !== false && reporting?.statefulSettings?.enabled) {
```

Comment:
> The problem seems to be that Core's deprecation checks aren't aware of the default value

I did not follow the specific problem you're facing, but again, the configuration deprecation system's only purpose is to warn the user about settings **defined in the config file** as being deprecated. The system not being aware of any arbitrary default, constraint, mutation, or anything else, defined in your plugin's configuration schema is normal and by design.

Reply:
Hi @pgayvallet I'm just running locally yarn es serverless and yarn serverless-es. When I run this on main currently in the kibana server logs this log appears that I'm hoping to prevent:
```
[2023-10-23T08:57:56.803-06:00][WARN ][config.deprecation] The default mechanism for Reporting privileges will work differently in future versions, which will affect the behavior of this cluster. Set "xpack.reporting.roles.enabled" to "false" to adopt the future behavior before upgrading.
```

Reply:
Then you need this `xpack.reporting.statefulSettings.enabled=true` setting to be explicitly defined in our serverless config file, for the reasons explained in my previous reply. 

But:

> the config value for roles.enabled setting is set to false by default in serverless

Same, this value is set at the schema's level, not the config level atm

https://github.com/elastic/kibana/blob/c7d2eb55ff324296f6817a3dfd718c88537f7f17/x-pack/plugins/reporting/server/config/schema.ts#L86-L90

So the correct approach would probably be to set `xpack.reporting.roles.enabled: false` in the serverless config file.

Reply:
I agree we should remedy this issue by updating [the serverless config file](https://github.com/elastic/kibana/blob/6a6b83e/config/serverless.yml#L1) with `xpack.reporting.statefulSettings.enabled: false` and perhaps `xpack.reporting.roles.enabled: false` as well.